pymc_marketing/clv/plotting.py

```diff
@@ -156,7 +158,7 @@ def _create_frequency_recency_meshes(
 
 
 def plot_frequency_recency_matrix(
-    model,
+    model: Union[BetaGeoModel, ParetoNBDModel],
```

Comment:
You can technically get away with anything that has data attribute and `expected_num_purchases` method 

If used typing.Protocol for this and python can figure out the rest

Might be something like 
```python
from typing import Protocol

class ExpectNumberPurchasesModel(Protocol): 
    @property 
    def data(self) -> pd.DataFrame: 
       """DataFrame with frequency and recency columns."""
    
    def expected_num_purchases(
        self,
        customer_id: Union[np.ndarray, pd.Series],
        t: Union[np.ndarray, pd.Series, TensorVariable],
        frequency: Union[np.ndarray, pd.Series, TensorVariable],
        recency: Union[np.ndarray, pd.Series, TensorVariable],
        T: Union[np.ndarray, pd.Series, TensorVariable],
    ) -> xr.DataArray:
        """Return the expected number of purchases based on the inputs."""

```

With this, we don't need to change the type hint with new models

Reply:
I wanted to indicate only these models are supported. In addition, `expected_num_purchases` is actually named `expected_purchases` in `ParetoNBDModel`, so I had to add a hotfix that specifically checks for this object instance. [BG/NBD and Pareto/NBD still have different APIs](https://github.com/pymc-labs/pymc-marketing/issues/527) and standardizing them will be a massive PR in itself. I want to get this bug fix merged ASAP.

Reply:
The hotfix will reduce testing coverage, but both models test out fine in the notebooks.